src/Core/src/Sts/Input/AssumeRoleRequest.php

```diff
@@ -186,9 +186,9 @@ public function getSerialNumber(): ?string
     }
 
     /**
-     * @return Tag[]
+     * @return Tag[]|null
      */
-    public function getTags(): array
+    public function getTags(): ?array
```

Comment:
At first I thought that `?array` was the right way to handle getters. It's consistent with all other getter.

But in that precise example, I think that'll reduce DX. leading to
```
if ($request->getTags() !== null) {
  foreach ($request->getTags() as $tag) {}
}

// or
foreach ($request->getTags() ?? [] as $tag) {}
```

I think we can safetly always fallback to `[]` , and change the signature of this getter to `array`.
If one day we change our mind, we'll be able to add a deprecation notice and change the behavior.

In DynamoDb, is it important to distinguished null from empty array (the original bug)?
- if yes, we should go with feature flag as suggested by @Nyholm (only on really required properties)
- if no, we just fallback to ` ?? []` everywhere

Reply:
> In DynamoDb, is it important to distinguished null from empty array (the original bug)?

from what I've seen of it, it's not a problem. Getter can continue to return empty array.

Reply:
great, feature flag behavior is not (yet) needed. internal call distinguish null from empty, but external signature always return array.

what do you think @Nyholm

Reply:
Lets try to use the fallback and we will see =)

Reply:
Just updated the PR with the fallback to array.

I still have some fix needed on 3 S3 operations but main modification is here.